### ISSUE TITLE
Typeahead select with keyboard.enter fires both selectItem and keydown.enter

### ISSUE THREAD

**Issue by mcescalante** [ng-bootstrap/ng-bootstrap#3070]
This is similar to #980 which is over 2 years old and is closed as resolved, but I am having the same problem with the latest versions.

### Bug description:

I have a typeahead that has `focusFirst` as false and redirects the user to a "search" page if enter is pressed and no typeahead result is selected. If I select a typeahead with my mouse, the expected behavior, `selectItem` is called. If I use the arrow keys on my keyboard and select a result and press enter, both `selectItem` and `keyup.enter` are fired when I would only like `selectItem` to fire.

Here is a GIF of me demoing both events firing followed by just `selectItem` firing when using the mouse to click in the below provided stackblitz:

![](https://cl.ly/b19fa32449ca/Screen%252520Recording%2525202019-03-14%252520at%25252003.05%252520PM.gif)

### Link to minimally-working StackBlitz that reproduces the issue:

https://stackblitz.com/edit/angular-6pwwyg

Note that the `selectItem` and `keypress.enter` events just log to console. See GIF above for demo of both events firing

### Versions of Angular, ng-bootstrap and Bootstrap:

Angular: 7.1.x

ng-bootstrap: 4.0.4

Bootstrap: 4.3.1

**Comment by maxokorokov** [comment]
Maybe I'm mistaken, but I can't see how this could be fixed, because your `(keyup.Enter)` listener is independent from the one set by the typeahead and is always executed first in my tests.

**Comment by mcescalante** [comment]
Yeah that's absolutely possible. Is there a different listener that would be better to use or a strategy to overcome this kind of behavior? For what it's worth, so far, I've "monkey patched" this in my app by checking what the actual model value is with an if/else in my `keyPress` function and then handling it appropriately (this works in my case).

Thanks for taking time to look at my weird edge case!

**Comment by AriPapari** [comment]
Any updates on this? Seems like a really weird behaviour deep inside the component. Trying to fix this on surface is really not beautiful.

**Comment by lazarljubenovic** [comment]
Since `ng-bootstrap` calls `preventDefault` on the `keydown` event (see below),

https://github.com/ng-bootstrap/ng-bootstrap/blob/b5d5febde5691286211dd2401c2cb5334db0851f/src/typeahead/typeahead.ts#L293-L294

you can check in your own handler for the `defaultPrevented` flag before you trigger your search function.
